src/pages/NewSensor/NewSensorStepThree.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable react/forbid-prop-types */
+import React from 'react';
+import { View } from 'react-native';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import { TabContainer } from './TabContainer';
+import {
+  CalendarIcon,
+  HazardIcon,
+  InfoIcon,
+  PaperSection,
+  FlexRow,
+  PageButton,
+  Spacer,
+  WithSpace,
+  TextWithIcon,
+  TextEditor,
+  EditorRow,
+  DateEditor,
+  DurationEditor,
+  TimeEditor,
+} from '../../widgets';
+
+import { WizardActions } from '../../actions/WizardActions';
+import { goBack } from '../../navigation/actions';
+import { selectNewSensor } from '../../selectors/newSensor';
+import { NewSensorActions } from '../../actions/index';
+import { useLoadingIndicator } from '../../hooks/useLoadingIndicator';
+import { DARKER_GREY, LIGHT_GREY, SUSSOL_ORANGE, WHITE } from '../../globalStyles';
+import { buttonStrings, vaccineStrings } from '../../localization';
+import { SECONDS } from '../../utilities/constants';
+
+export const NewSensorStepThreeComponent = ({
+  logInterval,
+  loggingDelay,
+  name,
+  code,
+  updateName,
+  connectToSensor,
+  updateCode,
+  updateLogInterval,
+  updateLoggingDelay,
+  exit,
+  previousTab,
+}) => {
+  const withLoadingIndicator = useLoadingIndicator();
+
+  return (
+    <TabContainer>
+      <WithSpace space={10} horizontal={false} vertical>
+        <PaperSection height={200} headerText={vaccineStrings.new_sensor_step_three_title}>
+          <EditorRow label={vaccineStrings.sensor_name} Icon={<InfoIcon color={DARKER_GREY} />}>
+            <TextEditor size="large" value={name} onChangeText={updateName} />
+            <TextEditor label={vaccineStrings.sensor_code} value={code} onChangeText={updateCode} />
+          </EditorRow>
+          <EditorRow
+            label={vaccineStrings.logging_interval}
+            Icon={<InfoIcon color={DARKER_GREY} />}
+          >
+            <DurationEditor
+              value={logInterval / SECONDS.ONE_MINUTE}
+              label=""
+              onChange={updateLogInterval}
+            />
+          </EditorRow>
+        </PaperSection>
+      </WithSpace>
+
+      <PaperSection height={130} contentContainerStyle={{ flex: 1 }}>
+        <EditorRow label={vaccineStrings.start_logging} Icon={<CalendarIcon color={DARKER_GREY} />}>
+          <DateEditor onPress={updateLoggingDelay} date={loggingDelay} />
+          <TimeEditor onPress={updateLoggingDelay} time={loggingDelay} />
+        </EditorRow>
+      </PaperSection>
+      <TextWithIcon left size="s" Icon={<HazardIcon color={LIGHT_GREY} />}>
```

Comment:
Using this button row in a few places, was going to extract but didn't? (Could also utilize in dispensing pages.. possibly adding it into the 'wizard' component, but have to consider validation rules and communicating from the child (tab) to the parent (wizard) when to allow moving forward etc)